python/geom.py

```diff
@@ -0,0 +1,173 @@
+                 H_chi3_diag=Vector3(),
+                 D_conductivity_diag=Vector3(),
+                 B_conductivity_diag=Vector3()):
+
+        self.epsilon_diag = epsilon_diag
+        self.epsilon_offdiag = epsilon_offdiag
+        self.mu_diag = mu_diag
+        self.mu_offdiag = mu_offdiag
+        self.E_susceptibilities = E_susceptibilities
+        self.H_susceptibilities = H_susceptibilities
+        self.E_chi2_diag = E_chi2_diag
+        self.E_chi3_diag = E_chi3_diag
+        self.H_chi2_diag = H_chi2_diag
+        self.H_chi3_diag = H_chi3_diag
+        self.D_conductivity_diag = D_conductivity_diag
+        self.B_conductivity_diag = B_conductivity_diag
+
+
+class Susceptibility:
+
+    def __init__(self, sigma_diag=Vector3(), sigma_offdiag=Vector3()):
+        self.sigma_diag = sigma_diag
+        self.sigma_offdiag = sigma_offdiag
+
+
+class LorentzianSusceptibility(Susceptibility):
+
+    def __init__(self, frequency=0.0, gamma=0.0, **kwargs):
+        super(LorentzianSusceptibility, self).__init__(**kwargs)
+        self.frequency = frequency
+        self.gamma = gamma
+
+
+class DrudeSusceptibility(Susceptibility):
+
+    def __init__(self, frequency=0.0, gamma=0.0, **kwargs):
+        super(DrudeSusceptibility, self).__init__(**kwargs)
+        self.frequency = frequency
+        self.gamma = gamma
+
+
+class NoisyLorentzianSusceptibility(LorentzianSusceptibility):
+
+    def __init__(self, noise_amp=0.0, **kwargs):
+        super(NoisyLorentzianSusceptibility, self).__init__(**kwargs)
+        self.noise_amp = noise_amp
+
+
+class NoisyDrudeSusceptibility(DrudeSusceptibility):
+
+    def __init__(self, noise_amp=0.0, **kwargs):
+        super(NoisyDrudeSusceptibility, self).__init__(**kwargs)
+        self.noise_amp = noise_amp
+
+
+class GeometricObject(object):
+
+    def __init__(self, material=Medium(), center=Vector3()):
+        self.material = material
+        self.center = center
+
+    def __contains__(self, point):
+        return mp.point_in_objectp(point, self)
+
+
+class Sphere(GeometricObject):
+
+    def __init__(self, radius=1.0, **kwargs):
+        self.radius = radius
+        super(Sphere, self).__init__(**kwargs)
+
+    @property
+    def radius(self):
+        return self._radius
+
+    @radius.setter
+    def radius(self, val):
+        if not val or non_negative(val):
+            self._radius = val
+        else:
+            raise ValueError("Radius cannot be negative. Got {}".format(val))
+
+
+class Cylinder(GeometricObject):
+
+    def __init__(self, axis=Vector3(0, 0, 1), radius=1.0, height=1.0, **kwargs):
+        self.axis = axis
+        self.radius = radius
+        self.height = height
+        super(Cylinder, self).__init__(**kwargs)
+
+    @property
+    def radius(self):
+        return self._radius
+
+    @property
+    def height(self):
+        return self._height
+
+    @radius.setter
+    def radius(self, val):
+        if non_negative(val) or not val:
```

Comment:
Since you have this check a lot, maybe it would be better to have a `check_nonnegative(name, val)` function that does:
```py
if val < 0:
    raise ValueError("{} cannot be negative: {}".format(name, val))
```
and then you can just do `self._radius = check_nonnegative("Cylinder.radius", val)`